### PR TITLE
Proper login error handling for deactivated users

### PR DESCRIPTION
**Pull request by jakubkuc**
I want to merge this change because error message and code should be proper for users that are not yet activated by email and deactivated.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4583 INVALID_CREDENTIALS instead of ACCOUNT_NOT_CONFIRMED error on login](https://app.clickup.com/t/2549495/SALEOR-4583)

File: saleor/graphql/account/tests/mutations/test_token_create.py [hunk 1]
```diff
@@ -109,3 +109,38 @@ def test_create_token_invalid_email(api_client, customer_user):
     response_error = content["data"]["tokenCreate"]["errors"][0]
     assert response_error["code"] == expected_error_code
     assert response_error["field"] == "email"
+
```
Comment (L3str4nge) [inline saleor/graphql/account/tests/mutations/test_token_create.py:112]: Do we have a test for scenario when user is active and has last login? If not then would be good to add it

File: saleor/graphql/account/mutations/authentication.py [hunk 2]
```diff
@@ -99,6 +99,25 @@ def get_user(cls, _info, data):
                     )
                 }
             )
+        if not user.is_active and not user.last_login:
+            raise ValidationError(
+                {
+                    "email": ValidationError(
+                        "Account needs to be confirmed via email",
```
Comment (IKarbowiak) [inline saleor/graphql/account/mutations/authentication.py:106]: 👮‍♀️ 
```suggestion
                        "Account needs to be confirmed via email.",
```

File: saleor/graphql/account/mutations/authentication.py [hunk 3]
```diff
@@ -99,6 +99,25 @@ def get_user(cls, _info, data):
...
+                        code=AccountErrorCode.ACCOUNT_NOT_CONFIRMED.value,
+                    )
+                }
+            )
+
+        if not user.is_active and user.last_login:
+            raise ValidationError(
+                {
+                    "email": ValidationError(
+                        "Account inactive",
```
Comment (IKarbowiak) [inline saleor/graphql/account/mutations/authentication.py:116]: 👮‍♀️ 
```suggestion
                        "Account inactive.",
```

**Review by IKarbowiak — APPROVED** [review]

**Review by tomaszszymanski129 — APPROVED** [review]

File: saleor/graphql/account/mutations/authentication.py (hunk 2 above)
Comment (jakubkuc) [reply saleor/graphql/account/mutations/authentication.py]: fixed

File: saleor/graphql/account/mutations/authentication.py (hunk 3 above)
Comment (jakubkuc) [reply saleor/graphql/account/mutations/authentication.py]: fixed

**Review by L3str4nge — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**